checkers/components/Game.tsx

```diff
@@ -0,0 +1,236 @@
+  return {
+    gameId: 'checkers',
+    gameLocalizeId: 'CHECKERS_GAME_NAME',
+    initialState: getInitialState(),
+    component: CheckersComponent,
+    riddleLevels,
+    getPossibleMoves,
+    getStateScoreForIndex0,
+    checkRiddleData,
+  };
+}
+
+const CheckersComponent: React.FunctionComponent<GameProps<IState>> = (props: GameProps<IState>) => {
+  const {appState} = useStoreContext();
+  const {move, setMove, showHint} = props;
+  const {turnIndex, state} = move;
+  const {riddleData, board /* miniMoves */} = state;
+
+  let hint: boolean | null = false;
+  let rowHint: boolean | null = false;
+  let ridRow: number;
+  let ridCol: number;
+
+  if (showHint && riddleData) {
+    if (riddleData.length == 1) {
+      ridRow = parseInt(riddleData[0], 10);
+      rowHint = true;
+    } else {
+      ridRow = parseInt(riddleData[0], 10);
+      ridCol = parseInt(riddleData[1], 10);
+      hint = true;
+    }
+  } else {
+    rowHint = false;
+    hint = false;
+  }
+
+  function showPossibleMoves() {
+    allPossibleMoves.forEach((element) => {
+      board[element.row][element.col] = 'SH';
+    });
+    setMove(move);
+  }
+  function removeSHFromBoard() {
+    allPossibleMoves.forEach((element) => {
+      board[element.row][element.col] = 'DS';
+    });
+  }
+
+  function clickedOn(row: number, col: number) {
+    if (riddleData && hint) {
+      board[ridRow][ridCol] = 'WM';
+    }
+    if (riddleData && rowHint) {
+      for (let i = 0; i < 7; i += 2) {
+        board[ridRow][i] = 'WM';
+      }
+    }
+    state.error = null;
+    try {
+      if (pieceIsSelected) {
+        pieceIsSelected = false;
+        removeSHFromBoard();
+        const toDelta = {col: col, row: row};
+        const fromDelta = {col: selectedMovingPiece.col, row: selectedMovingPiece.row};
+        const Nmove = createMiniMove(board, fromDelta, toDelta, turnIndex);
+        setMove(Nmove);
+      }
+      if (isOwnColor(turnIndex, getColor(board[row][col]))) {
+        pieceIsSelected = true;
+        selectedMovingPiece = {col: col, row: row};
+        allPossibleMoves = getAllPossibleMoves(board, selectedMovingPiece, turnIndex);
+        showPossibleMoves();
+      }
+    } catch (e) {
+      state.error = e.message;
+      const errorUpdate = {endMatchScores: move.endMatchScores, turnIndex: move.turnIndex, state: move.state};
+      state.error = localize(e.message, appState);
+      console.log('error:', state.error, 'appstate:', appState);
+      setMove(errorUpdate);
+    }
+  }
+
+  const rows = [0, 1, 2, 3, 4, 5, 6, 7];
+  const cols = [0, 1, 2, 3, 4, 5, 6, 7];
+
+  function getPiece(r: number, c: number) {
+    if (board[r][c] == '--') return null;
+    if (board[r][c] == 'DS') return null;
+    let piece;
+    if (hint) {
+      board[ridRow][ridCol] = 'WMH';
+    }
+    if (rowHint) {
+      for (let i = 0; i < 7; i += 2) {
+        board[ridRow][i] = 'WMH';
+      }
+    }
+    switch (board[r][c]) {
```

Comment:
I like how the file names are making it easy to understand what each string means.
Like when you look at the WMH and then read that the file name is white_man_hint.
I am just not sure what the cro stands for, is it crown?